txstratum/filters.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) Hathor Labs and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Any, Set
+
+from hathorlib.scripts import P2PKH
+from structlog import get_logger
+
+if TYPE_CHECKING:
+    from hathorlib import BaseTransaction
+
+    from txstratum.toi_client import TOIAsyncClient
+
+
+logger = get_logger()
+
+
+class TXFilter(ABC):
+    """Base class for tx filters."""
+
+    @abstractmethod
+    async def check_tx(self, tx: 'BaseTransaction', data: Any) -> bool:
+        """Return if the tx should be blocked."""
+        raise NotImplementedError
+
+
+class FileFilter(TXFilter):
+    """Filter tx based on a set of banned tx_ids and addresses."""
+
+    def __init__(self, banned_tx_ids: Set[bytes] = set(), banned_addresses: Set[str] = set()) -> None:
+        """Init filter."""
+        self.log = logger.new()
+        self.banned_tx_ids = banned_tx_ids
```

Comment:
It seems like a static filter. I feel it should receive a file and read from there. We can even detect a file change and update this list automatically without a restart (or we can do it in the future).

Reply:
It is a static filter. I think the idea of watching changes on a file and loading them at runtime is great, but it should be done on it's own PR.

Reply:
I agree. Would you open an issue with a brief description, please?